src/Meshes/Meshes.jl

```diff
@@ -132,4 +132,98 @@ struct EquiangularCubedSphereMesh{FT} <: AbstractMesh{FT}
     n::Int64
 end
 
+"""
+    Mesh2D{I,IA2D,FT,FTA2D} <: AbstractMesh{FT}
+
+Conformal mesh for a 2D manifold. The manifold can be 
+embedded in a higher dimensional space.
+
+                        Quadrilateral
+
+                v3            f4           v4
+                  o------------------------o
+                  |                        |		  face    vertices
+                  |                        |             
+                  |                        |		   f1 =>  v1 v3 
+               f1 |                        | f2        f2 =>  v2 v4
+                  |                        |		   f3 =>  v1 v2
+                  |                        |           f4 =>  v3 v4
+                  |                        |
+                  |                        |
+                  o------------------------o
+                 v1           f3           v2
+
+z-order numbering convention for 2D quadtrees
+
+Reference:
+
+p4est: SCALABLE ALGORITHMS FOR PARALLEL ADAPTIVE
+MESH REFINEMENT ON FORESTS OF OCTREES∗
+CARSTEN BURSTEDDE†, LUCAS C. WILCOX‡ , AND OMAR GHATTAS§
+SIAM J. Sci. Comput. Vol. 33, No. 3, pp. 1103-1133
+
+https://p4est.github.io/papers/BursteddeWilcoxGhattas11.pdf
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct Mesh2D{I, IA1D, IA2D, FT, FTA2D} <: AbstractMesh{FT}
+    "# of unique nodes in the mesh"
+    nnodes::I
+    "# of unique faces in the mesh"
+    nfaces::I
+    "# of elements in the mesh"
+    nelems::I
+    "# of zones in the mesh"
+    nbndry::I
+    "x₁, x₂, ... coordinates of nodes `(nnodes, dim)`, dim can be greater than 2 for 2D manifolds embedded in higher dimensional space"
+    coordinates::FTA2D
+    "face node numbers `(nfaces, 2)`"
+    face_nodes::IA2D
+    "boundary elems for each face `(nfaces, 2)`"
+    face_neighbors::IA2D
+    "face zones for each face `(nfaces, 1)`"
+    face_bndry::IA1D
+    "node numbers for each elem `(nelems, 4)`"
+    elem_nodes::IA2D
```

Comment:
I think these `nnodes` and `elem_nodes` can lead to confusion. We have vertex/vertices for the corners of a quadrilateral in a mesh. Nodes are the Finite Element nodes, i.e., there are p+1 of them, for a polynomial degree p finite element space. But the corners of a quad face are called vertices. See the API for [`vertex_coordinates`](https://github.com/CliMA/ClimaCore.jl/blob/f24aa819613346197cc4e6a8123a60b62ebf6b22/src/Topologies/Topologies.jl#L56)